### ISSUE TITLE
time.Sleep() never returns

### ISSUE THREAD

**Issue by zhaytee** [golang/go#16382]
This wasn't happening on my machine before I installed the macOS beta. Kind of regretting it now! 😁 

1. What version of Go are you using?
`go version go1.6.2 darwin/amd64`

2. What operating system and processor architecture are you using?
`macOS Sierra beta2 (macOS 10.12 16A238m)`
`on amd64`

3. What did you do?
Compiled and ran the following small program:
```
package main

import (
	"fmt"
	"time"
)

func main() {
	i := 0
	for {
		i++
		fmt.Printf("Sleeping, pass #%d... ", i)
		time.Sleep(time.Millisecond * 50)
		fmt.Printf("done.\n")
	}
}
```

4. What did you expect to see?
A practically infinite number of loop iterations.

5. What did you see instead?
Output for a few seconds, then nothing, because time.Sleep() fails to return after a few iterations of the loop.
Using a debugger (https://github.com/derekparker/delve), I'm able to interrupt the program and grab a stack trace when it freezes, and it's always exactly the same:
```
0  0x0000000000056c43 in runtime.mach_semaphore_timedwait
   at /usr/local/go/src/runtime/sys_darwin_amd64.s:421
1  0x0000000000026d0b in runtime.semasleep1
   at /usr/local/go/src/runtime/os1_darwin.go:412
2  0x000000000004b250 in runtime.semasleep.func1
   at /usr/local/go/src/runtime/os1_darwin.go:439
3  0x0000000000053597 in runtime.systemstack
   at /usr/local/go/src/runtime/asm_amd64.s:291
4  0x000000000002d390 in runtime.startTheWorldWithSema
   at /usr/local/go/src/runtime/proc.go:986
5  0x0000000000183600 in (nil)
   at ?:-1
```

I get the same result if the for loop is moved into a separate goroutine, or if the sleep duration is varied.

**Comment by cespare** [comment]
Duplicate of #16272

**Comment by mikioh** [comment]
Merged into #16354.

**Comment by bradfitz** [comment]
Dup of #16272
